auslib/test/admin/views/test_releases.py

```diff
@@ -382,22 +451,26 @@ def testLocaleGet404(self):
         ret = self._get("/releases/c/builds/h/u")
         self.assertStatusCode(ret, 404)
 
-    def testLocaleRevertsPartialUpdate(self):
-        data = json.dumps(dict(complete=dict(filesize=1)))
-        with mock.patch('auslib.global_state.dbo.releases.addLocaleToRelease') as r:
-            r.side_effect = Exception("Fail")
-            ret = self._put('/releases/a/builds/p/l', data=dict(data=data, product='a', version='c', data_version=1, schema_version=1))
-            self.assertStatusCode(ret, 500)
-            ret = dbo.releases.t.select().where(dbo.releases.name == 'a').execute().fetchone()
-            self.assertEqual(ret['product'], 'a')
-            self.assertEqual(ret['version'], 'a')
-            self.assertEqual(json.loads(ret['data']), dict(name='a', hashFunction="sha512", schema_version=1))
+    # FIXME: We shouldn't rely on 500 to validate behaviour. This test should fake a 400 instead.
```

Comment:
Don't praise me too much :). I'm not going to fix this as part of this patch unless you insist.